app/models/service_template_ansible_playbook.rb

```diff
@@ -0,0 +1,79 @@
+class ServiceTemplateAnsiblePlaybook < ServiceTemplateGeneric
+  def self.default_provisioning_entry_point(_service_type)
+    '/Service/Generic/StateMachines/GenericLifecycle/provision'
+  end
+
+  def self.default_retirement_entry_point
+    '/Service/Generic/StateMachines/GenericLifecycle/retire'
+  end
+
+  # create ServiceTemplate and supporting ServiceResources and ResourceActions
+  # options
+  #   :name
+  #   :description
+  #   :service_template_catalog
+  #   :config_info
+  #     :provision
+  #       :service_dialog_id (or)
+  #       :new_dialog_name
+  #       :variables
+  #       :hosts
+  #       :credentials
+  #       :playbook_id
+  #     :retirement (same as provision)
+  #     :reconfigure (same as provision)
+  #
+  def self.create_catalog_item(options)
+    options      = options.merge(:service_type => 'atomic', :prov_type => 'generic_ansible_playbook')
+    service_name = options[:name]
+    description  = options[:description]
+    config_info  = validate_config_info(options)
+
+    transaction do
+      create(options.except(:config_info)).tap do |service_template|
+        [:provision, :retirement, :reconfigure].each do |action|
+          prepare_job_template_and_dialog(action, service_name, description, options) if config_info.key?(action)
+        end
+        service_template.create_resource_actions(config_info)
+      end
+    end
+  end
+
+  def self.prepare_job_template_and_dialog(action, service_name, description, config_info)
+    job_template = create_job_template("#{service_name}_#{action}", description, config_info[action])
```

Comment:
We should consider naming the job template name with something that can't change like `"MiqService_#{id}_#{action}"`.